src/docs/sphinx/dev_guide/gpu_porting.rst

```diff
@@ -161,6 +160,199 @@ GPU device, or on both a GPU device and a CPU host. For example::
   When Axom is built without RAJA, ``axom::for_all`` becomes a ``for``-loop on
   host (CPU).
 
+%%%%%%%%%%%%%%%%
+Portability
+%%%%%%%%%%%%%%%%
+
+Adherence to the GPU porting guidelines generally results in code that will compile and run on
+multiple backends. However, backends such as CUDA require additional guidelines.
+
+Do not use generic lambda functions (auto parameters) with ``axom::for_all``
+or the code will not compile under nvcc.
+
+Do this:
+
+  .. code-block:: cpp
+
+      axom::for_all<ExecSpace>(n, AXOM_LAMBDA(axom::IndexType index) { /* body */});
+
+
+Do NOT do this:
+
+  .. code-block:: cpp
+
+      axom::for_all<ExecSpace>(n, AXOM_LAMBDA(auto index) { /* body */});
+
+
+Data in Axom is often contained in useful containers such as ``axom::Array``.
```

Comment:
Suggest starting the paragraph of each rule with a few bold summary words.  So perhaps 
1. start line 170 with **No auto lambda parameters.** 
2. start this line with **Pass ArrayView by value.**  
3. start line 229 with **No for_all from private/protected.**
4. start line 264 with **Avoid for_all from within lambda.**
5. on line 313, put the first sentence in bold.
6. start line 317 with **Specialize templates outside other classes.**